File: test/GenerateStreamParkPassword.java
import org.apache.shiro.crypto.hash.Sha256Hash;

/**
 * StreamPark 密码生成工具
 * 使用 Shiro 的 SHA-256 加密算法生成密码哈希
 */
public class GenerateStreamParkPassword {
    
    public static void main(String[] args) {
        // 密码
        String password = "streampark";
        
        // Salt (26 字符)
        String salt = "26f87aee40e022f38e8ca2d8c9";
        
        // 使用 Shiro SimpleHash 生成密码哈希
        // 算法: SHA-256
        // 迭代次数: 1024
        Sha256Hash hash = new Sha256Hash(password, salt, 1024);
        String hashedPassword = hash.toHex();
        
        System.out.println("========================================");
        System.out.println("StreamPark 密码生成");
        System.out.println("========================================");
        System.out.println();
        System.out.println("输入:");
        System.out.println("  密码: " + password);
        System.out.println("  Salt: " + salt);
        System.out.println("  算法: SHA-256");
        System.out.println("  迭代: 1024");
        System.out.println();
        System.out.println("输出:");
        System.out.println("  Password Hash: " + hashedPassword);
        System.out.println();
        System.out.println("SQL 更新语句:");
        System.out.println("  UPDATE t_user SET");
        System.out.println("    salt = '" + salt + "',");
        System.out.println("    password = '" + hashedPassword + "'");
        System.out.println("  WHERE username = 'admin';");
        System.out.println();
    }
}

File: test/generate_streampark_password.py
#!/usr/bin/env python3
"""
StreamPark 密码生成工具
使用 SHA-256 + PBKDF2 生成密码哈希
"""

import hashlib

def sha256_hash_iterations(password, salt, iterations=1024):
    """
    使用 SHA-256 进行多次迭代哈希
    模拟 Shiro SimpleHash 的行为
    """
    # 第一次哈希: SHA-256(password + salt)
    hash_value = hashlib.sha256((password + salt).encode('utf-8')).digest()
    
    # 迭代哈希
    for i in range(1, iterations):
        hash_value = hashlib.sha256(hash_value).digest()
    
    # 转换为十六进制字符串
    return hash_value.hex()

def main():
    password = "streampark"
    salt = "26f87aee40e022f38e8ca2d8c9"
    iterations = 1024
    
    print("=" * 50)
    print("StreamPark 密码生成")
    print("=" * 50)
    print()
    print("输入:")
    print(f"  密码: {password}")
    print(f"  Salt: {salt}")
    print(f"  算法: SHA-256")
    print(f"  迭代: {iterations}")
    print()
    
    # 生成密码哈希
    hashed_password = sha256_hash_iterations(password, salt, iterations)
    
    print("输出:")
    print(f"  Password Hash: {hashed_password}")
    print()
    print("SQL 更新语句:")
    print(f"  UPDATE t_user SET")
    print(f"    salt = '{salt}',")
    print(f"    password = '{hashed_password}'")
    print(f"  WHERE username = 'admin';")
    print()

if __name__ == "__main__":
    main()
